download_and_create_reference_datasets/v03/write_clinvar_ht.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+import argparse
+import os
+import tempfile
+
+import hail as hl
+
+from hail_scripts.reference_data.constants import GCS_PREFIXES
+from hail_scripts.utils.clinvar import (
+    download_and_import_latest_clinvar_vcf,
+    parsed_clnsig,
+    parsed_clnsigconf,
+    CLINVAR_ASSERTIONS_LOOKUP,
+    CLINVAR_DEFAULT_PATHOGENICITY,
+    CLINVAR_PATHOGENICITIES_LOOKUP,
+    CLINVAR_GOLD_STARS_LOOKUP,
+)
+from hail_scripts.utils.hail_utils import write_ht
+
+CLINVAR_HT_PATH = 'clinvar/clinvar.GRCh{genome_version}.{timestamp}.ht'
+PARTITIONS = 100 # per https://github.com/broadinstitute/seqr-loading-pipelines/pull/383
+
+def run(environment: str):
+    for genome_version in ['37', '38']:
+        with tempfile.NamedTemporaryFile(suffix='.vcf.gz') as tmp_file:
+            mt = download_and_import_latest_clinvar_vcf(genome_version, tmp_file)
+            timestamp = hl.eval(mt.version)
+            ht = mt.rows()
+            ht = ht.select(
+                alleleId=ht.info.ALLELEID,
+                pathogenicities_id=CLINVAR_PATHOGENICITIES_LOOKUP.get(
+                    parsed_clnsig(ht)[0], 
+                    CLINVAR_PATHOGENICITIES_LOOKUP[CLINVAR_DEFAULT_PATHOGENICITY],
+                ),
+                assertions_ids=(
+                    parsed_clnsig(ht)
+                    .filter(lambda x: ~CLINVAR_PATHOGENICITIES_LOOKUP.contains(x))
+                    .map(lambda x: CLINVAR_ASSERTIONS_LOOKUP[x])
+                ),
+                conflictingPathogenicities_ids=(
+                    parsed_clnsigconf(ht)
+                    .starmap(lambda pathogenicity, _: CLINVAR_PATHOGENICITIES_LOOKUP[pathogenicity])
+                ),
+                conflictingPathogenicities_counts=(
+                    parsed_clnsigconf(ht)
+                    .starmap(lambda _, count: count)
+                ),
+                goldStars=CLINVAR_GOLD_STARS_LOOKUP.get(hl.delimit(ht.info.CLNREVSTAT)),
+            ).annotate_globals(
+                enum_definitions=hl.dict({
+                    'pathogenicities_id': CLINVAR_PATHOGENICITIES_LOOKUP,
```

Comment:
since by definition the script that is reading this table needs to do a reverse lookup, I think it might be more straightforward to put the array of options into the global instead of the dictionary lookup

Reply:
Yep!  I had the same idea here: https://github.com/broadinstitute/seqr-loading-pipelines/pull/440#discussion_r1171862792

I like that!